docpages/example_programs/the_basics/using_callback_functions.md

```diff
@@ -0,0 +1,95 @@
+    bot.on_slashcommand([&bot](const dpp::slashcommand_t& event) -> void {
+        if (event.command.get_command_name() == "msgs-get") {
+            int64_t limit = std::get<int64_t>(event.get_parameter("quantity"));
+
+            /* get messages using ID of the channel the command was issued in */
+            bot.messages_get(event.command.channel_id, 0, 0, 0, limit, [event](const dpp::confirmation_callback_t& callback) -> void {
+                if (callback.is_error()) { /* catching an error to log it */
+                    std::cout << callback.get_error().message << std::endl;
+                    return;
+                }
+
+                auto messages = callback.get<dpp::message_map>();
+                /* std::get<dpp::message_map>(callback.value) would give the same result */
+                
+                std::string contents;
+                for (const auto& x : messages) { /* here we iterate through the dpp::message_map we got from callback... */
+                    contents += x.second.content + '\n'; /* ...where x.first is ID of the current message and x.second is the message itself. */
+                }
+
+                event.reply(contents); /* we will see all those messages we got, united as one! */
+            });
+        } else if (event.command.get_command_name() == "channel-create") {
+
+            /* create a text channel */
+            dpp::channel channel = dpp::channel()
+                .set_name("test")
+                .set_guild_id(event.command.guild_id);
+
+            bot.channel_create(channel, [&bot, event](const dpp::confirmation_callback_t& callback) -> void {
+                if (callback.is_error()) { /* catching an error to log it */
+                    bot.log(dpp::loglevel::ll_error, callback.get_error().message);
+                }
+
+                auto channel = callback.get<dpp::channel>();
+                /* std::get<dpp::channel>(callback.value) would give the same result */
+
+                /* reply with the created channel information */
+                dpp::message message = dpp::message("The channel's name is `" + channel.name + "`, ID is `" + std::to_string(channel.id) + " and type is `" + std::to_string(channel.get_type()) + "`.");
+                /* note that channel types are represented as numbers */
+                event.reply(message);
+            });
+        } else if (event.command.get_command_name() == "msg-error") {
+            bot.message_get(0, 0, [event](const dpp::confirmation_callback_t& callback) -> void {
+                /* the error will occur since there is no message with ID '0' that is in a channel with ID '0' (I'm not explaining why) */
+                if (callback.is_error()) {
+                    event.reply(callback.get_error().message);
+                    return;
+                }
+
+                /* we won't be able to get here because of the return; statement */
+                auto message = callback.get <dpp::message>();
```

Comment:
`get<dpp::message>` without the space please 😄

Reply:
Please, I wanna sleep 😭